apps/builder/app/builder/shared/navigator-tree.tsx

```diff
@@ -115,9 +116,33 @@ export const NavigatorTree = () => {
       shallowEqual(selectedInstanceSelectorStore.get(), instanceSelector) ===
       false
     ) {
+      const previousInstanceSelector = selectedInstanceSelectorStore.get();
+      if (previousInstanceSelector) {
+        publish({
+          type: "emitComponentHook",
+          payload: {
+            name: "onNavigatorDeselect",
```

Comment:
nit: onNavigatorUnselect would be slightly better, it means one specific thing, where deselect is more like deselect everything

In the future we will have ability to select multiple items in navigator

Reply:
maybe even better onNavigatorItemUnselect

Reply:
or alternatively we will have to use argument to distinguish between unselecting/selecting multiple items vs one

Reply:
anyways, up to you, just so you are aware of interactions we will need in the future: select one, select multiple, select all, and correspondingly unselect for each type